### ISSUE TITLE
DK-8KZ not supported

### ISSUE THREAD

**Issue by bjacobse** [camrein/EzGraver#12]
DK-8KZ isn't working for me when I use Ubuntu/Linux. (I have just tried with Win7 and your program, can upload and start burning. with win7 laser engraving with DK-8KZ working fine, I did also install the driver that can be downloaded from the neje homepage)
. I was looking forward to be able to use my Linux laptop with your nice work :-)
I think the firmware is different for the DK-8KZ and 'other' Neje older laser engravers. I don't know how I can debug the USB/serial communication protocol.

I can notice that ttyUSB0 is being found when using EzgraverUI and it seems that I can connect. but DK-8KZ isn't moving, uploading or anything

Br
Brian

**Comment by camrein** [comment]
Hi Brian

First of all, glad you like that little piece software.

Good you tested it on Win7. As it appears that EzGraver works as expected (you can upload and start the engrave process), it is compatible to your engraver. Therefore, we can exclude the idea that there is a compatiblity issue with your model and/or firmware.

But this also means that the issue must be somewhere else. Just to double check the possibility that EzGraver actually recognizes your engraver and not any other device (as of now, it simply detects any USB device providing a serial port).
Could you simply fire-up EzGraver without having DK-8KZ connected to your computer? If it still shows up ttyUSB0 within the dropdown it means that its some other device. Therefore, the installation installation of the appropriate for your device is necessary.

Another possibility is that another software is blocking the access to the device. But I think my first suggestion is a good starting point.

Cheers
Christoph

**Comment by bjacobse** [comment]
Hi Cristoph,
Thank you for your quick reply, I wasn't actual sure how I could reach you

When using your program in Linux/Ubuntu and nothing is attached to USB
port, I can see "all" tty adn why trying to connect, I get an serial can't
connect error - This is as it should.

See my dmesg, USB serial is detected.
starting Ezgraver, shows ONLY tty USB0 -this is also fine. I can connect,
and it looks quite fine.

I use Ezgraver in prompt, and have attached the "log"

I wonder if the windows serial driver has some translation? since I have
also tried this guys script without luck
https://github.com/AxelTB/nejePrint/blob/master/README.md

Br
Brian


On Sat, Nov 12, 2016 at 10:32 PM, Christoph Amrein <notifications@github.com

**Comment by camrein** [comment]
Hi Brian

I don't see your attachement. I suppose that's because you have responded by mail.
Quickly trying to connect on an Ubuntu VM showed me a connection error as well (I didn't need to install a driver either).
```bash
./EzGraverCli
> Available Ports: ttyUSB0

./EzGraverCli r ttyUSB0
> instantiating EzGraver on port "ttyUSB0"
> failed to establish connection on port "ttyUSB0"
> "Permission denied"
```
As it can be seen, I was receiving a"permission denied" error. So I tryed it with extended privileges and succeeded in sending commands to my engraver.
```bash
sudo ./EzGraverCli r ttyUSB0
> instantiating EzGraver on port "ttyUSB0"
> resetting
> transmitting 1 bytes: "f9"
```

Of course, running software unnecessarily as root is usually not a good idea. So best is, adding your user to the respective group. A reboot was required after adding my user to the dialout group.
```bash
sudo adduser camrein dialout
> Adding user 'camrein' to group 'dialout' ...
> Adding user camrein to group dialout
> Done.
```
In my case (I believe this is generally the case for ubuntu) it was the group "dialout".
But to be sure, you can simply check the file-handle:
```bash
ls -al /dev/ttyUSB0
> crw-rw---- 1 root dialout 188, 0 Nov 13 11:27 /dev/ttyUSB0
```
I hope this fixes your issue as well.

PS: I have noticed an issue that the CLI version of EzGraver does not exit after transmitting the command (on Linux). So for now, force-closing with CTRL+C is probably required.

Best regards
Christoph

**Comment by bjacobse** [comment]
hi Christoph
Its working now on my Linux/Ubuntu :-)

I was wondering about rights so I reompilet with sudo.
sudo qmake -qt=qt5 -config release EzGraver.pro
sudo make
sudo make install

run the program as my normal user
EzGraverUi/EzGraverUi

and now it's working, I can set home, and set center, upload, preview and
burn

Thank you for your support.
You are BTW missing an icon on your page, a donate button :-)

I am happy to use your sw. It's same on windows ans Linux, and also the
antivirus i + firefox sn't happy the official Neje sw that can be
downloaded from trusfer.com to be used from windows.

-Brian




On Sun, Nov 13, 2016 at 11:41 AM, Christoph Amrein <notifications@github.com

**Comment by camrein** [comment]
Hi Brian

Great that it is working now for you. :D 
But thank you as well. You're probably not the only one having this issue, so this ticket will surely help them.

Feel free to contact me again.
About the donate button. I might add one in the future. :P

About the original software. I had the same concerns and I thought it could be fun developing a cross-platform application in C++.

So, all the best.
Christoph

**Comment by bjacobse** [comment]
Hi Christoph,
Thank you for your support

I'ts not a shame to add a donate button :-)

I would have added some text in case of troubles the possibility to use
sudo command, but I'm not able. I hope someone can see the closed issue,
since I think that github is hiding it

Best regards
Brian

**Comment by peterlaszlo** [comment]
Hi Cristoph!

My Neje dk8-kz engraver wont work with ezgraver on linux mint.
I can connect on ttyUSB0 fine, but only upload working. I cannot use Home, Start, Reset, Down, etc.
It works fine on win7, but I usually working on Linux Mint.

**Comment by camrein** [comment]
Hi

First of all, I probably have to clarify that being able to connect to a device that is listed in the connection drop-down does not necessarily guarantee that the correct device is recognized. Just to make sure as I heard this situation in the past. =)
So what I mean, is that I guess that you're not connected to the correct device as EzGraver does currently not run any checks if it's the right device. So the first thing I'd like to ask you if the device ttyUSB0 still does show up when your engraver is disconnected. If that is the case, your ttyUSB0 serial device is something else (maybe a virtual one).
So if you can confirm that the ttyUSB0 device is NOT your engraver, you'll need to install the proper drivers. Unfortunately, I have only tested EzGraver with ubuntu which appears to be shipping with the necessary components; I cannot really tell you what to install.

Cheers
Christoph

**Comment by peterlaszlo** [comment]
Hi Cristoph,
Thank you for your fast reply!
I checked all my usb. My other ttyUSB are my two other reprap printer. Detached all usb devices and external drives, 3d printers. How can I install the proper Neje drivers on linux?
I installed the reprap machine drivers 4 years ago but I forgot all the procedure.

**Comment by camrein** [comment]
So I tried EzGraver with a VirtualBox Linux Mint image.
If I remember right, I came across the same situation as with Ubuntu:

Without the proper access rights, EzGraver will print a permission denied error:
![image](https://user-images.githubusercontent.com/18049310/30771373-b42d0544-a046-11e7-8805-609f942c90a2.png)

When executing it with root access, everything worked as expected. However, as you initially said you could connect to the device, so I don't think it's related to a permission error.
![image](https://user-images.githubusercontent.com/18049310/30771376-ca948c58-a046-11e7-81fa-d3cfb3ee6923.png)

I didn't have to install anything besides the build tools, i.e., no additional drivers were necessary. The image I was playing around with was a "Linux Mint 18.2 Cinnamon (32bit)" one, just in case they are shipping the necessary drivers with more recent versions.

All commands, except the moving commands, should work on either protocol version (v1 and v2 dropdown). As I know that DK-8-KZ uses the v2 protocol, please make sure you have selected such. Nonetheless, the commands such as start and stop should work with both.

One last thing you could try, if you didn't manage to get it running by now, is installing the Arduino IDE. With some luck, it could add the missing dependencies:
https://playground.arduino.cc/Linux/Mint

**Comment by Jazzjerry** [comment]
Hello Camrein,

I absolutely respect your effort and other peoples same mindmap about making the cheap NEJE laser engravers accessible to the Linux users. This is where this type of diy machines should originate.

So I only use linux, and saw this real cheap machine and had a look to see if I would be able to get it running and found some backwards engineering... some scripts and your perfect effort to actually get this machine running on my mint machine...

So I ordered my NEJE DK-8-KZ build by Trustfer hoping I would be able to get it to burn my idears.

And so far... It has burned the test butterfly that was preinstalled on its eprom many times...

I cant seem to get the data connection needed to actually send a new Image 512x512 px to the eprom.

Your EzGraver 3.2 software recognizes the serial connection of my NEJE DK-8-KZ perfectly.

So I try and connect.

connecting to port ttyUSB0 with protocol version 2
connection established successfully
loading image: /#####/####/Downloads/ezgraver source/EzGraver-3.2.0/screenshot.png
erasing EEPROM
uploading image to EEPROM
upload completed

So all seems ok up to this point.

The Engraver wakes up after uploading my image through your Gui or Cli..

It buzzes and relocates after a few seconds.

Then I try and start...

Nothing happens.

Not through the Gui nor Cli..

The up down left right reset start pause etc ... buttons have no effect.

When I press the red button on the machine... I starts printing from its eprom and gues what...

It again prints the preinstalled test butterfly NEJE put on the eprom.

I would love to help us out with pentesting and trying to figure out why I cant communicate with the machine. Maybe the protocol has changed. 

I tried V1 and V2 of your protocol options that appear automaticly within EzGraver but untill now no result nor any change.

I truely hope you are still reading this and willing to help the diy people on a china budget out to get this working open source.

Sorry to cause this trouble... but hope it might be simple and just something I have overlooked.

I truly hope I will be able to engrave some nuts and autumn nature with poetry on a small scale this year ;-)

I love your work so far. Looks perfect. Hopefully I can get it to work for me.
